packages/block-editor/src/components/rich-text/index.js

```diff
@@ -1176,51 +1173,57 @@ RichText.defaultProps = {
 
 const RichTextContainer = compose( [
 	withInstanceId,
-	withBlockEditContext( ( context, ownProps ) => {
-		// When explicitly set as not selected, do nothing.
-		if ( ownProps.isSelected === false ) {
-			return {
-				clientId: context.clientId,
-			};
-		}
-		// When explicitly set as selected, use the value stored in the context instead.
-		if ( ownProps.isSelected === true ) {
-			return {
-				isSelected: context.isSelected,
-				clientId: context.clientId,
-			};
-		}
-
-		// Ensures that only one RichText component can be focused.
-		return {
-			isSelected: context.isSelected && context.focusedElement === ownProps.instanceId,
-			setFocusedElement: context.setFocusedElement,
-			clientId: context.clientId,
-		};
-	} ),
-	withSelect( ( select ) => {
+	withBlockEditContext( ( { clientId } ) => ( { clientId } ) ),
+	withSelect( ( select, {
+		clientId,
+		instanceId,
```

Comment:
Is it expected to be allow to pass the `identifier` as a prop to `RichText` component? Otherwise, we could use `instanceId` as is in both `withSelect` and `withDispatch`.

Reply:
Just `instanceId` would work, but I thought it might be good to "name" it if we can. E.g. one could programmatically select a piece of text in a paragraph why knowing the block client ID and RichText identifier. No one could know the instance ID... I'm thinking we could eventually require the identifier to be the same as the block attribute key it uses.